### DISCUSSION TITLE
Custom Boundary Rect

### DISCUSSION THREAD

**Discussion by vavilon224** [fabricjs/fabric.js#9764] [Q&A]
Hello everyone, help. Any ideas for a solution? @ShaMan123 
Here is my code

```
export const boundaryRectGenerator = (object) => {
  if (object) {
    const boundaryRect = object.getBoundingRect();

    const scaleX = object.scaleX || 1;
    const scaleY = object.scaleY || 1;

    const scaleFactor = 1.35;

    const newWidth = boundaryRect.width * scaleX * scaleFactor;
    const newHeight = boundaryRect.height * scaleY * scaleFactor;

    object.set({
      width: newWidth,
      height: newHeight,
    });

    object.setCoords();
  }
};
```

The goal is to increase the current boundary rect taking into account Scale
I need this because my text path goes out of bounds
![_240325190132](https://github.com/fabricjs/fabric.js/assets/87662684/1c4b7661-4d49-442a-b505-5e531521c6a1)

**Comment by vavilon224** [comment]
I want to say thank you in advance, you always help) @ShaMan123

**Comment by ShaMan123** [comment]
I don't understand what your desired behavior is

**Reply by vavilon224** [reply]
I want after applying to my text path.
Increase boundary Rect.

**Reply by vavilon224** [reply]
![_240326091704](https://github.com/fabricjs/fabric.js/assets/87662684/036cb0f3-31fb-4ad0-855f-ff2018cd96d4)
It’s also strange that if I do textPath.toDataUrl()
it doesn't return me base64 string
@ShaMan123

**Comment by ShaMan123** [comment]
Seems to me you also need to set left and top.
If I were you I would consider using bezier-js to calculate the bbox of the path using https://pomax.github.io/bezierjs/#offset to offset via font size or the highest char and https://pomax.github.io/bezierjs/#bbox
